server/integration-test/src/test/java/com/oceanbase/odc/service/structurecompare/mysql/DefaultDBStructureComparatorTest.java

```diff
@@ -0,0 +1,136 @@
+import com.oceanbase.odc.service.structurecompare.model.ComparisonResult;
+import com.oceanbase.odc.service.structurecompare.model.DBObjectComparisonResult;
+import com.oceanbase.odc.service.structurecompare.model.DBStructureComparisonConfig;
+import com.oceanbase.odc.test.database.TestDBConfiguration;
+import com.oceanbase.odc.test.database.TestDBConfigurations;
+import com.oceanbase.odc.test.util.FileUtil;
+import com.oceanbase.tools.dbbrowser.model.DBObjectType;
+
+/**
+ * @author jingtian
+ * @date 2024/9/14
+ */
+public class DefaultDBStructureComparatorTest extends PluginTestEnv {
+    private static final String BASE_PATH =
+            "src/test/resources/structurecompare/mysql/";
+    private static String sourceSchemaDdl = FileUtil.loadAsString(BASE_PATH + "source_schema_ddl.sql");
+    private static String targetSchemaDdl = FileUtil.loadAsString(BASE_PATH + "target_schema_ddl.sql");
+    private static final String sourceDrop = FileUtil.loadAsString(BASE_PATH + "source_drop.sql");
+    private static final String targetDrop = FileUtil.loadAsString(BASE_PATH + "target_drop.sql");
+    private static TestDBConfiguration configuration = TestDBConfigurations.getInstance().getTestMysqlConfiguration();
+    private static JdbcTemplate jdbcTemplate = new JdbcTemplate(configuration.getDataSource());
+    private final static String sourceSchemaName = generateSchemaName() + "_source";
+    private final static String targetSchemaName = generateSchemaName() + "_target";
+    private static TestDBConfiguration srcConfiguration = new TestDBConfiguration();
+    private static TestDBConfiguration tgtConfiguration = new TestDBConfiguration();
+    private static DefaultDBStructureComparator comparator = new DefaultDBStructureComparator();
+    private static List<DBObjectComparisonResult> results;
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @BeforeClass
+    public static void setUp() throws SQLException {
+        jdbcTemplate.execute("create database `" + sourceSchemaName + "`");
+        jdbcTemplate.execute("create database `" + targetSchemaName + "`");
+
+        BeanUtils.copyProperties(configuration, srcConfiguration);
+        srcConfiguration.setDefaultDBName(sourceSchemaName);
+        srcConfiguration.initDataSource();
+
+        BeanUtils.copyProperties(configuration, tgtConfiguration);
+        tgtConfiguration.setDefaultDBName(targetSchemaName);
+        tgtConfiguration.initDataSource();
+
+        jdbcTemplate.execute("use `" + sourceSchemaName + "`");
+        jdbcTemplate.execute(sourceDrop);
+        jdbcTemplate.execute(sourceSchemaDdl);
+        jdbcTemplate.execute("use `" + targetSchemaName + "`");
+        jdbcTemplate.execute(targetDrop);
+        jdbcTemplate.execute(targetSchemaDdl);
+        jdbcTemplate.execute("use `" + configuration.getDefaultDBName() + "`");
+
+        results = comparator.compare(getSourceConfig(srcConfiguration),
```

Comment:
under test method should called in test case, not in setUp

Reply:
In fact, all compare methods of the structure comparison class are implemented in setUp. This is because we only need to read the ddl script once and compare it once to obtain the comparison results of all unit tests, which can also improve the efficiency of unit testing.

Reply:
the basic concept is wrong here

Reply:
if only once comparing, the only one test case

Reply:
but each test case is written for a different test scenario, and it is not appropriate to write all of them in one test method. I think just consider the test class for structure comparison  as a special scenario, which compares all table structures at once.